backend/src/main/java/org/eclipse/tractusx/puris/backend/delivery/logic/service/DeliveryService.java

```diff
@@ -224,67 +227,43 @@ protected List<String> validateOwnResponsibility(Delivery delivery) {
-                            ) && (
-                                delivery.getDestinationBpna() == null || 
-                                partnerSite.get().getAddresses().stream().anyMatch(address -> address.getBpna().equals(delivery.getDestinationBpna())) 
-                            )) {
-                            return new ArrayList<>();
+                        supplierPathErrors = validateLocationsAsSupplier(delivery, "supplier", ownSites, partnerSites);
+                        if (supplierPathErrors.isEmpty()) {
+                            valid = true;
                         }
                     }
                     if (delivery.getMaterial().isMaterialFlag()) {
-                        ownSite = ownSites.stream().filter(site -> site.getBpns().equals(delivery.getDestinationBpns())).findFirst();
-                        partnerSite = partnerSites.stream().filter(site -> site.getBpns().equals(delivery.getOriginBpns())).findFirst();
-                        if (ownSite.isPresent() && partnerSite.isPresent() && (
-                                delivery.getDestinationBpna() == null || 
-                                ownSite.get().getAddresses().stream().anyMatch(address -> address.getBpna().equals(delivery.getDestinationBpna()))
-                            ) && (
-                                delivery.getOriginBpna() == null || 
-                                partnerSite.get().getAddresses().stream().anyMatch(address -> address.getBpna().equals(delivery.getOriginBpna())) 
-                            )) {
-                            return new ArrayList<>();
+                        customerPathErrors = validateLocationsAsCustomer(delivery, "customer", ownSites, partnerSites);
+                        if (customerPathErrors.isEmpty()) {
+                            valid = true;
+                        }
+                    }
+                    if (!valid) {
+                        errors.add(String.format("Responsibility conditions for material '%s' for partial responsibility (incoterm '%s') are not met. Either origin site bpns '%s' does not match to own configured sites or destination site bpns '%s' does not match to configured sites for partner '%s'. Additionally this behavior might not be applicable to the material configuration as product (%b) or material (%b).", delivery.getMaterial().getOwnMaterialNumber(), delivery.getIncoterm().getValue(), delivery.getOriginBpns(), delivery.getDestinationBpns(), delivery.getPartner().getBpnl(), delivery.getMaterial().isProductFlag(), delivery.getMaterial().isMaterialFlag()));
+                        if (delivery.getMaterial().isProductFlag() && !supplierPathErrors.isEmpty()) {
+                            errors.addAll(supplierPathErrors);
+                        }
+                        if (delivery.getMaterial().isMaterialFlag() && !customerPathErrors.isEmpty()) {
+                            errors.addAll(customerPathErrors);
                         }
```

Comment:
```suggestion
                        errors.addAll(supplierPathErrors);
                        errors.addAll(customerPathErrors);
```
If the errors are empty, we don't get further errors. Thus, this should be more resilient without. Now it led to not being shown.